internal/credspropagation/openstack.go

```diff
@@ -0,0 +1,273 @@
+// Copyright 2024
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package credspropagation
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	texttemplate "text/template"
+
+	"gopkg.in/yaml.v3"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	capo "sigs.k8s.io/cluster-api-provider-openstack/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	hmc "github.com/Mirantis/hmc/api/v1alpha1"
+)
+
+// Fields needed to populate the cloud.conf template
+type CloudConfFields struct {
```

Comment:
Can we use `AuthOptions` struct available in https://github.com/gophercloud/gophercloud/blob/v2.4.0/auth_options.go instead? Just a thought to consider.

Reply:
I did see this before. But it has only the auth fields. We would need to do a mix of importing this from gopher where maintaining our struct for the rest of the fields. That's why I chose to go with our custom struct so that its clean.

Reply:
i believe the API is stable, so supporting the own struct won't hurt in this case, considering introducing a new dependency for a mere struct, so I personally would agree with @bnallapeta on this and leave the struct

Reply:
I got it. Thank you!